{"hierarchy":{"paths":[["doc:\/\/codes.soto.soto-core\/documentation\/SotoCore","doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/RetryPolicy"]]},"identifier":{"url":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/RetryPolicy\/getRetryWaitTime(error:attempt:)","interfaceLanguage":"swift"},"abstract":[{"text":"Returns whether we should retry and how long we should wait before retrying","type":"text"}],"sections":[],"schemaVersion":{"patch":0,"major":0,"minor":3},"metadata":{"symbolKind":"method","externalID":"s:8SotoCore11RetryPolicyP03getC8WaitTime5error7attemptAA0C6StatusOSgs5Error_p_SitF","title":"getRetryWaitTime(error:attempt:)","required":true,"modules":[{"name":"SotoCore"}],"fragments":[{"kind":"keyword","text":"func"},{"kind":"text","text":" "},{"text":"getRetryWaitTime","kind":"identifier"},{"text":"(","kind":"text"},{"kind":"externalParam","text":"error"},{"text":": ","kind":"text"},{"preciseIdentifier":"s:s5ErrorP","text":"Error","kind":"typeIdentifier"},{"kind":"text","text":", "},{"text":"attempt","kind":"externalParam"},{"kind":"text","text":": "},{"kind":"typeIdentifier","text":"Int","preciseIdentifier":"s:Si"},{"kind":"text","text":") -> "},{"kind":"typeIdentifier","text":"RetryStatus","preciseIdentifier":"s:8SotoCore11RetryStatusO"},{"text":"?","kind":"text"}],"role":"symbol","roleHeading":"Instance Method"},"primaryContentSections":[{"declarations":[{"languages":["swift"],"tokens":[{"text":"func","kind":"keyword"},{"text":" ","kind":"text"},{"kind":"identifier","text":"getRetryWaitTime"},{"text":"(","kind":"text"},{"kind":"externalParam","text":"error"},{"kind":"text","text":": "},{"kind":"typeIdentifier","text":"Error","preciseIdentifier":"s:s5ErrorP"},{"kind":"text","text":", "},{"text":"attempt","kind":"externalParam"},{"text":": ","kind":"text"},{"kind":"typeIdentifier","text":"Int","preciseIdentifier":"s:Si"},{"kind":"text","text":") -> "},{"kind":"typeIdentifier","text":"RetryStatus","identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/RetryStatus","preciseIdentifier":"s:8SotoCore11RetryStatusO"},{"text":"?","kind":"text"}],"platforms":["macOS"]}],"kind":"declarations"},{"parameters":[{"name":"error","content":[{"inlineContent":[{"text":"Error returned by HTTP client","type":"text"}],"type":"paragraph"}]},{"name":"attempt","content":[{"type":"paragraph","inlineContent":[{"text":"Retry attempt number","type":"text"}]}]}],"kind":"parameters"}],"variants":[{"traits":[{"interfaceLanguage":"swift"}],"paths":["\/documentation\/sotocore\/retrypolicy\/getretrywaittime(error:attempt:)"]}],"kind":"symbol","references":{"doc://codes.soto.soto-core/documentation/SotoCore/RetryPolicy":{"title":"RetryPolicy","kind":"symbol","navigatorTitle":[{"kind":"identifier","text":"RetryPolicy"}],"identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/RetryPolicy","url":"\/documentation\/sotocore\/retrypolicy","fragments":[{"text":"protocol","kind":"keyword"},{"text":" ","kind":"text"},{"text":"RetryPolicy","kind":"identifier"}],"role":"symbol","type":"topic","abstract":[{"type":"text","text":"Protocol for Retry strategy. Has function returning amount of time before the next retry after an HTTP error"}]},"doc://codes.soto.soto-core/documentation/SotoCore/RetryPolicy/getRetryWaitTime(error:attempt:)":{"kind":"symbol","type":"topic","identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/RetryPolicy\/getRetryWaitTime(error:attempt:)","abstract":[{"text":"Returns whether we should retry and how long we should wait before retrying","type":"text"}],"title":"getRetryWaitTime(error:attempt:)","required":true,"fragments":[{"text":"func","kind":"keyword"},{"text":" ","kind":"text"},{"text":"getRetryWaitTime","kind":"identifier"},{"text":"(","kind":"text"},{"text":"error","kind":"externalParam"},{"text":": ","kind":"text"},{"preciseIdentifier":"s:s5ErrorP","text":"Error","kind":"typeIdentifier"},{"text":", ","kind":"text"},{"kind":"externalParam","text":"attempt"},{"text":": ","kind":"text"},{"kind":"typeIdentifier","preciseIdentifier":"s:Si","text":"Int"},{"kind":"text","text":") -> "},{"text":"RetryStatus","kind":"typeIdentifier","preciseIdentifier":"s:8SotoCore11RetryStatusO"},{"text":"?","kind":"text"}],"role":"symbol","url":"\/documentation\/sotocore\/retrypolicy\/getretrywaittime(error:attempt:)"},"doc://codes.soto.soto-core/documentation/SotoCore":{"kind":"symbol","type":"topic","abstract":[{"text":"The core framework for Soto, a Swift SDK for AWS","type":"text"}],"title":"SotoCore","identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore","url":"\/documentation\/sotocore","role":"collection"},"doc://codes.soto.soto-core/documentation/SotoCore/RetryStatus":{"role":"symbol","identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/RetryStatus","kind":"symbol","fragments":[{"text":"enum","kind":"keyword"},{"kind":"text","text":" "},{"text":"RetryStatus","kind":"identifier"}],"abstract":[{"type":"text","text":"Return value for "},{"type":"codeVoice","code":"RetryPolicy.getRetryWaitTime"},{"text":". Either retry after time amount or donâ€™t retry","type":"text"}],"type":"topic","url":"\/documentation\/sotocore\/retrystatus","navigatorTitle":[{"text":"RetryStatus","kind":"identifier"}],"title":"RetryStatus"}}}